core-bundle/src/Resources/contao/library/Contao/System.php

```diff
@@ -400,10 +399,14 @@ public static function getReferer($blnEncodeAmpersands=false, $strTable=null)
 
 				list($path, $query) = explode('?', $url, 2);
 
-				$queryObj = new Query($query);
-				$queryObj = $queryObj->withoutPairs($params);
+				parse_str($query, $pairs);
 
-				return $path . $queryObj->getUriComponent();
+				foreach ($params as $param)
+				{
+					unset($pairs[$param]);
+				}
+
```

Comment:
Should we return without the `?` if the whole query string was unset?
```suggestion

				if (!$pairs) 
				{
					return $path;
				}
```

Reply:
Merged in cbae1469f9fa05aa3672f68e518ad86cdd77d435.